src/facade/dragonfly_connection.cc

```diff
@@ -1418,15 +1418,23 @@ void Connection::ShutdownSelf() {
   util::Connection::Shutdown();
 }
 
-void Connection::Migrate(util::fb2::ProactorBase* dest) {
+bool Connection::Migrate(util::fb2::ProactorBase* dest) {
   // Migrate is used only by replication, so it doesn't have properties of full-fledged
   // connections
   CHECK(!cc_->async_dispatch);
-  CHECK_EQ(cc_->subscriptions, 0);    // are bound to thread local caches
-  CHECK_EQ(self_.use_count(), 1u);    // references cache our thread and backpressure
-  CHECK(!dispatch_fb_.IsJoinable());  // can't move once it started
+  CHECK_EQ(cc_->subscriptions, 0);  // are bound to thread local caches
+  CHECK_EQ(self_.use_count(), 1u);  // references cache our thread and backpressure
+  if (dispatch_fb_.IsJoinable()) {  // can't move once it started
```

Comment:
what is triggering the dispatch_fb_ creation on shutdown?

Reply:
This https://github.com/dragonflydb/dragonfly/blob/main/src/facade/dragonfly_listener.cc#L301 which calls `SendCheckpoint`. See also: https://github.com/dragonflydb/dragonfly/blob/main/src/facade/dragonfly_listener.cc#L301 and https://github.com/dragonflydb/dragonfly/blob/main/src/facade/dragonfly_listener.cc#L455 

Moreover, this is what I was telling you about migrations. In this flow, we migrate and it happens that the traversal fiber happens before the migration and so when we migrate the dispatch fiber is already started. Now, if we remove this check and cancel the migration, it could also be the case that another fiber run which killed the socket (after PreShutdown and DispatchTracker) and I have seen this being triggered and that's why I checked for the socket to be active. 

Of course, it could be the case that traversal happened `after migration` but before the connection is relinked to the thread local and in that case we would probably deadlock or loose some important message being send from the traversal.